### PR TITLE
881 [Solutions] Update Search Payload

### LINKED ISSUE #881: [Solutions] Update Search Payload
Need to update payload for Search Query.
Add parseInt() to documentPosition value in Click analytics call

### PR DESCRIPTION
**Pull request by sharanagoudapatil**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #881 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://881-update-search-payload--danaher-ls-aem--hlxsites.hlx.page/us/en/solutions/mabs/cell-line-development/products

### CODE REVIEW

**Review by davidzochowski — CHANGES_REQUESTED** [review]
@sharanagoudapatil You updated the analytics payload. This change is for the Search payload.

**Comment by sharanagoudapatil** [comment]
@davidzochowski - I've updated Search payload. Please have a look. Thanks.

File: scripts/commerce.js [hunk 1]
```diff
@@ -127,15 +127,40 @@ function getWorkflowFamily() {
...
+  const searchHistory = searchHistoryString ? JSON.parse(searchHistoryString) : [];
   const payload = {
+    actionCause: 'interfaceLoad',
+    actionsHistory: searchHistory.map(({ time, value, name }) => ({ time, value, name })),
+    anonymous: false,
+    aq: `@${qParam}==${wfPath}`,
+    customData: {
+      context_workflow: `${wfPath}`,
+      context_host: window.DanaherConfig.host,
+      context_internal: false,
```
Comment (davidzochowski) [inline scripts/commerce.js:142]: This value should be based off of the exclude-from-analytics cookie. The same as what is being used for the other payloads.

File: scripts/commerce.js [hunk 2]
```diff
@@ -127,15 +127,40 @@ function getWorkflowFamily() {
...
+    },
     context: {
       workflow: `${wfPath}`,
       host: `${host}`,
       internal: false,
```
Comment (davidzochowski) [inline scripts/commerce.js:147]: This value should be based off of the exclude-from-analytics cookie. The same as what is being used for the other payloads.

File: scripts/commerce.js [hunk 3]
```diff
@@ -127,15 +127,40 @@ function getWorkflowFamily() {
...
     },
-    aq: `@${qParam}==${wfPath}`,
+    documentReferrer: document.referrer,
+    documentLocation: window.location.href,
+    firstResult: 0,
+    language: 'en',
```
Comment (davidzochowski) [inline scripts/commerce.js:152]: set this field to 'locale' instead of 'language'

**Review by davidzochowski — DISMISSED** [review]
Requested changes

File: scripts/commerce.js [hunk 4]
```diff
@@ -127,15 +127,40 @@ function getWorkflowFamily() {
...
+  const searchHistoryString = localStorage.getItem('__coveo.analytics.history');
+  const searchHistory = searchHistoryString ? JSON.parse(searchHistoryString) : [];
   const payload = {
+    actionCause: 'interfaceLoad',
+    actionsHistory: searchHistory.map(({ time, value, name }) => ({ time, value, name })),
+    anonymous: false,
+    aq: `@${qParam}==${wfPath}`,
+    customData: {
+      context_workflow: `${wfPath}`,
+      context_host: window.DanaherConfig.host,
```
Comment (davidzochowski) [inline scripts/commerce.js:141]: Some code cleanup, you're already creating a host variable on line 129. could use that or remove the variable and use the config variable for line 146

File: scripts/commerce.js [hunk 5]
```diff
@@ -127,15 +127,40 @@ function getWorkflowFamily() {
 function getProductsOnSolutionsApiPayload(qParam) {
   const wfPath = getWorkflowFamily();
   const host = window.DanaherConfig !== undefined ? window.DanaherConfig.host : '';
+  const userTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+  const clientId = getCookie('coveo_visitorId');
+  const searchHistoryString = localStorage.getItem('__coveo.analytics.history');
+  const searchHistory = searchHistoryString ? JSON.parse(searchHistoryString) : [];
   const payload = {
```
Comment (davidzochowski) [inline scripts/commerce.js:134]: Missing the 'analytics' block. Some of the fields should be within that block.

File: scripts/commerce.js (hunk 1 above)
Comment (sharanagoudapatil) [reply scripts/commerce.js]: Done.

File: scripts/commerce.js (hunk 3 above)
Comment (sharanagoudapatil) [reply scripts/commerce.js:153]: updated.

File: scripts/commerce.js (hunk 4 above)
Comment (sharanagoudapatil) [reply scripts/commerce.js:142]: modified.

File: scripts/commerce.js (hunk 5 above)
Comment (sharanagoudapatil) [reply scripts/commerce.js:135]: updated.

File: scripts/commerce.js (hunk 2 above)
Comment (sharanagoudapatil) [reply scripts/commerce.js]: updated.

**Comment by davidzochowski** [comment]
Checking Coveo analytics. There are two issues:
1. Change logic so that if clientid is null, do NOT send analytics payload
 Update line 198  if (fullResponse && fullResponse.results.length > 0 **&& clientId !== null**) {
2. For onClickCoveoAnalyticsPayload, need to add customData.contentIDValue and customData.contentIDKey
Logic is same as line 76+77 in delayed.js
      context_contentIDValue: window.location.href,
      context_contentIDValue: 'permanentid' //this can be hardcoded

**Review by davidzochowski — APPROVED** [review]

**Pull request merged**